rasa/shared/exceptions.py

```diff
@@ -77,3 +82,19 @@ class InvalidConfigException(ValueError, RasaException):
 
 class UnsupportedFeatureException(RasaCoreException):
     """Raised if a requested feature is not supported."""
+
+
+class SchemaValidationError(RasaException, jsonschema.ValidationError):
+    """Raised if schema validation via `jsonschema` failed."""
+
+
+class InvalidEntityFormatException(RasaException):
```

Comment:
Should this also inherit from some kind of json error?

Reply:
can you detail why do you suggest this?

Reply:
Because of the comment: "InvalidEntityFormatException if provided entity is not valid json."
I would generally assume that if data (e.g. json or yaml) is an incorrect format i.e. cannot be parsed, then it would be some kind of decode, parse or value error. 
And if it can be parsed but does not match our schema, then it would be a Rasa exception.
Does this make sense? Or do we see `RasaException` as broader than that?

Reply:
`RasaException` is a base class for expected exceptions raised when using Rasa Open Source (and most probably resulting from a wrong usage). These exceptions do not get sent to Sentry (there is nothing we can do about it from an engineering POV) but are reported to the users CLI. I'll update the docstring of the exception to reflect this, I think it's missing. Does that make sense?

`InvalidEntityFormatException` -> I think it's a good point too. I'll see if I can easily inherit the [`JSONDecodeError`](https://docs.python.org/3/library/json.html#json.JSONDecodeError) (constructor requires more arguments). It's also better in terms of backwards-compatibility (if you expect the code to raise `JSONDecodeError`, it'll still work)

Reply:
changes in ece18bcfe17 and edcd70fae23